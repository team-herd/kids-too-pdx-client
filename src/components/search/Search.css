.Select {
  text-align: center;
  padding: .5em;

  select {
    display: block;
    margin: auto;
    padding: .5em;
    font-size: .8em;
  }

  input {
    margin: auto;
    text-align: center;
    display: block;
    width: 4em;
    padding: .5em;
  }

  label {
    font-weight: 800;
  }
}
  
@media only screen and (min-width: 1000px) {
  .Select {
    display: flex;
    justify-content: space-around;
    margin: 1em;

    label {
      display: flex;
      justify-content: center;
      margin: 0;
      justify-content: center
    }
    
    select {
      margin: 0;
    }
  }
}
 