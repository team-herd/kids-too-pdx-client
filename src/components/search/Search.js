import React from 'react';
import styles from './Search.css';

function Search() {
  return (
    <section className={styles.Select}>
      <select>
        <option value="DEFAULT" hidden>Category</option>
        <option value="sports">Sports</option>
        <option value="arts">Art</option>
        <option value="music">Music</option>
        <option value="performance">Performance</option>
        <option value="education">Educational</option>
        <option value="volunteer">Volunteer</option>
      </select>
      <label>
        Price:
        <input type="number" value="price" placeholder="max"/>
      </label>
      <label>
        Age:
        <input type="number" value="ageMax" placeholder="min age"/>
        <input type="number" value="ageMax" placeholder="max age"/>
      </label>
    </section>
  );
}

export default Search;
