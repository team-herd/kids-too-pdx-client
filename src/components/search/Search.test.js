import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

describe('Search', () => {
  it('snapshots Search', () => {
    const props = {
      onChange: jest.fn(),
      onSubmit: jest.fn()
    };

    const tree = renderer.create(
      <Search {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
