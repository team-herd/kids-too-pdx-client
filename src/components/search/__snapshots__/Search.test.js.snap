// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshots Search 1`] = `
<form
  className="Select"
  onSubmit={[MockFunction]}
>
  <select
    name="category"
    onChange={[MockFunction]}
  >
    <option
      hidden={true}
      value="DEFAULT"
    >
      Category
    </option>
    <option
      value="Sports"
    >
      Sports
    </option>
    <option
      value="Art"
    >
      Art
    </option>
    <option
      value="Music"
    >
      Music
    </option>
    <option
      value="Performance"
    >
      Performance
    </option>
    <option
      value="Educational"
    >
      Educational
    </option>
    <option
      value="Volunteer"
    >
      Volunteer
    </option>
  </select>
  <label>
    Price:
    <input
      name="price"
      onChange={[MockFunction]}
      placeholder="max"
      type="number"
    />
  </label>
  <label>
    Age:
    <input
      name="ageMin"
      onChange={[MockFunction]}
      placeholder="min age"
      type="number"
    />
    <input
      name="ageMax"
      onChange={[MockFunction]}
      placeholder="max age"
      type="number"
    />
  </label>
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
